### ISSUE TITLE
[Feature Request] Cache upcoming episodes

### ISSUE THREAD

**Issue by Shockshwat** [Stremio/stremio-core#761]
Specially with RD gone, many times episodes take a long time to load and once they are fully cached, there should be a setting which allows us to automatically start caching the next episode so as to provide a seamless experience when binging through shows.

**Comment by mark-addams001881** [comment]
Hello, we tried to solve the issue.

This is what we did:

Add a new setting for auto-caching next episode and implement the logic to start caching when the current episode is fully loaded.

You can review changes in this commit: https://github.com/mark-addams001881/Stremio-stremio-core-761/commit/b3ad9d423f7ae2d5784144fa4eb52e6e34b90804.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

Latta AI seeks to solve problems in open source projects as part of its mission to support developers around the world. Learn more about our mission at https://latta.ai/ourmission . If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by farebyting** [comment]
Ehh ,Isn't this the job of whatever torrentio addons or scraper you use?

**Comment by Shockshwat** [comment]
No, I meant to say the local cacheing or "downloading" of episodes as you stream. For example, if you are watching episode N of something and it gets downloaded 100% then it should start downloading the N+1 episode so that if the source is slow, you can start watching the next episode directly without that waiting to start new episode thing. I specially notice this with obscure shows and torbox isnt much of a help either as of now.

**Comment by farebyting** [comment]
Wow great idea!

**Comment by kKaskak** [comment]
outdated, duplicate
